### PR TITLE
Fix WaPo v3 indexing

### PR DESCRIPTION
**Pull request by PepijnBoers**
This PR resolves https://github.com/castorini/anserini/issues/1291 by making the `published_date` attribute optional for Washington Post articles during indexing. We do, however, attempt to obtain the publication date from the `contents` section when no dedicated `published_date` is available.

### CODE REVIEW

**Comment by lintool** [comment]
hey @PepijnBoers does this work for v2 also? We should probably maintain backwards compatibility?

File: src/main/java/io/anserini/collection/WashingtonPostCollection.java [hunk 1]
```diff
@@ -48,8 +48,8 @@
 
 /**
  * An instance of the <a href="https://trec.nist.gov/data/wapost/">TREC Washington Post Corpus</a>.
- * The collection contains 608,180 news articles and blog posts from January 2012 through August 2017,
```
Comment (lintool) [inline src/main/java/io/anserini/collection/WashingtonPostCollection.java:51]: Should we reference stats for v2 and v3 separately? I believe v1 is defunct, right?

Comment (PepijnBoers) [reply src/main/java/io/anserini/collection/WashingtonPostCollection.java:51]: We could mention that version 3 adds 154,418 new documents from 2018 and 2019 and removes (near) duplicates.

v2 was used in TREC News 2018 & 2019 and is basically v1 without exact duplicates.

**Comment by PepijnBoers** [comment]
This still works for v2 because all articles there have an explicit `published_date` attribute. The script always uses the attribute whenever it is available and only uses the date in `contents` if that is the only one present.

I have also built the v2 index and checked if I obtained the same scores for the `BM25+RM3` run for TREC 2019 bg-linking as were mentioned [here](https://github.com/castorini/anserini/blob/master/docs/regressions-backgroundlinking19.md).

**Comment by lintool** [comment]
great, thanks! I'll confirm regressions on my end and then we'll get this merged!

**Comment by lintool** [comment]
These regressions confirmed:

```
python src/main/python/run_regression.py --collection core18 --index
python src/main/python/run_regression.py --collection backgroundlinking18
python src/main/python/run_regression.py --collection backgroundlinking19
```

File: src/main/java/io/anserini/collection/WashingtonPostCollection.java (hunk 1 above)
Comment (lintool) [reply src/main/java/io/anserini/collection/WashingtonPostCollection.java:51]: Yes, I think we should explicitly describe the date ranges and num of documents of v2 and v3.

**Pull request merged**